Potential Stochastic Test Failure in `test_generate_sorting_to_inject`

This only failed on one OS/one python version. Not sure if this is a problem in generate_sorting_to_inject which can cause some failures. We can leave this issue open to monitor if this problem happens multiple times.

From action: https://github.com/SpikeInterface/spikeinterface/actions/runs/9898102287/job/27344129816?pr=3185

```
_______________________ test_generate_sorting_to_inject ________________________

    def test_generate_sorting_to_inject():
        durations = [10.0, 20.0]
        sorting = generate_sorting(num_units=10, durations=durations, sampling_frequency=30000, firing_rates=1.0)
>       injected_sorting = generate_sorting_to_inject(
            sorting, [int(duration * sorting.sampling_frequency) for duration in durations]
        )

src/spikeinterface/core/tests/test_generate.py:549: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
src/spikeinterface/core/generate.py:293: in generate_sorting_to_inject
    min_diff = np.min(np.abs(injected_spike_train[:, None] - spike_train[None, :]), axis=1)
/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/numpy/core/fromnumeric.py:2953: in min
    return _wrapreduction(a, np.minimum, 'min', axis, None, out,
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

obj = array([], shape=(0, 0), dtype=int64), ufunc = <ufunc 'minimum'>
method = 'min', axis = 1, dtype = None, out = None
kwargs = {'initial': <no value>, 'keepdims': <no value>, 'where': <no value>}
passkwargs = {}

    def _wrapreduction(obj, ufunc, method, axis, dtype, out, **kwargs):
        passkwargs = {k: v for k, v in kwargs.items()
                      if v is not np._NoValue}
    
        if type(obj) is not mu.ndarray:
            try:
                reduction = getattr(obj, method)
            except AttributeError:
                pass
            else:
                # This branch is needed for reductions like any which don't
                # support a dtype.
                if dtype is not None:
                    return reduction(axis=axis, dtype=dtype, out=out, **passkwargs)
                else:
                    return reduction(axis=axis, out=out, **passkwargs)
    
>       return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
E       ValueError: zero-size array to reduction operation minimum which has no identity
```

Another random failure in python 3.9 on Windows. From https://github.com/SpikeInterface/spikeinterface/actions/runs/9897996103/job/27343788820?pr=3180
```
FAILED src/spikeinterface/preprocessing/tests/test_silence.py::test_silence - AssertionError: assert 0.1109969 < 0.1
 +  where 0.1109969 = <built-in method sum of numpy.ndarray object at 0x000002B600FB4E70>()
 +    where <built-in method sum of numpy.ndarray object at 0x000002B600FB4E70> = array([0.08974379, 0.02125311], dtype=float32).sum
 +      where array([0.08974379, 0.02125311], dtype=float32) = <ufunc 'absolute'>((array([0.90464187, 0.9732841 ], dtype=float32) - array([0.99438566, 0.99453723], dtype=float32)))
 +        where <ufunc 'absolute'> = np.abs
 +        and   array([0.90464187, 0.9732841 ], dtype=float32) = <function std at 0x000002B64658BBF0>(array([[-0.11603655, -0.73109096],\n       [ 0.8392864 , -0.5543756 ],\n       [ 0.3452874 ,  0.68456554],\n       ...,\n       [-1.1869012 , -0.74847776],\n       [-0.66878515, -0.20611839],\n       [-0.93528956,  1.155699  ]], dtype=float32), axis=0)
 +          where <function std at 0x000002B64658BBF0> = np.std
= 1 failed, 2269 passed, 42 skipped, 749 deselected, 54 warnings in 114.73s (0:01:54) =

```

--- a/src/spikeinterface/core/tests/test_generate.py
+++ b/src/spikeinterface/core/tests/test_generate.py
@@ -547,7 +547,7 @@ def test_generate_sorting_to_inject():
     durations = [10.0, 20.0]
     sorting = generate_sorting(num_units=10, durations=durations, sampling_frequency=30000, firing_rates=1.0)
     injected_sorting = generate_sorting_to_inject(
-        sorting, [int(duration * sorting.sampling_frequency) for duration in durations]
+        sorting, [int(duration * sorting.sampling_frequency) for duration in durations], seed=2308
     )
     num_spikes = sorting.count_num_spikes_per_unit()
     num_injected_spikes = injected_sorting.count_num_spikes_per_unit()

--- a/src/spikeinterface/preprocessing/tests/test_silence.py
+++ b/src/spikeinterface/preprocessing/tests/test_silence.py
@@ -16,7 +16,7 @@ def test_silence(create_cache_folder):
 
     rec = generate_recording()
 
-    rec0 = silence_periods(rec, list_periods=[[[0, 1000], [5000, 6000]], []], mode="zeros")
+    rec0 = silence_periods(rec, list_periods=[[[0, 1000], [5000, 6000]], []], mode="zeros", seed=2308)
     rec0.save(verbose=False)
     traces_in0 = rec0.get_traces(segment_index=0, start_frame=0, end_frame=1000)
     traces_in1 = rec0.get_traces(segment_index=0, start_frame=5000, end_frame=6000)
@@ -25,7 +25,7 @@ def test_silence(create_cache_folder):
     assert np.all(traces_in1 == 0)
     assert not np.all(traces_out0 == 0)
 
-    rec1 = silence_periods(rec, list_periods=[[[0, 1000], [5000, 6000]], []], mode="noise")
+    rec1 = silence_periods(rec, list_periods=[[[0, 1000], [5000, 6000]], []], mode="noise", seed=2308)
     rec1 = rec1.save(folder=cache_folder / "rec_w_noise", verbose=False, overwrite=True)
     noise_levels = get_noise_levels(rec, return_scaled=False)
     traces_in0 = rec1.get_traces(segment_index=0, start_frame=0, end_frame=1000)

